### PR TITLE
Improve image UX

### PR DESCRIPTION
**Pull request by ericpromislow**
* After a build or pull command finishes successfully, clear the field
* Trim whitespace off ends of UI field values
* Refresh the list of images immediately after a command that modifies the underlying list

Signed-off-by: Eric Promislow <epromislow@suse.com>

### CODE REVIEW

File: background.ts [hunk 1]
```diff
@@ -181,10 +181,17 @@ Electron.ipcMain.on('confirm-do-image-deletion', async(event, imageName, imageID
...
 
   if (choice === 0) {
-    await imageManager.deleteImage(imageID);
-    refreshImageList();
+    try {
+      await imageManager.deleteImage(imageID);
+      refreshImageList();
+      event.reply('kim-process-ended', 0);
+    } catch (err) {
+      Electron.dialog.showMessageBox({
```
Comment (mook-as) [inline background.ts:189]: Do we not need a reply in this case?

File: src/components/Images.vue [hunk 2]
```diff
@@ -234,28 +235,33 @@ export default {
...
       this.startRunningCommand('delete');
-      ipcRenderer.send('confirm-do-image-deletion', obj.imageName, obj.imageID);
+      ipcRenderer.send('confirm-do-image-deletion', obj.imageName.trim(), obj.imageID.trim());
     },
     doPush(obj) {
       this.kimRunningCommand = `push ${ obj.imageName }:${ obj.tag }`;
+      this.refreshImagesOnEnd = false;
       this.startRunningCommand('push');
-      ipcRenderer.send('do-image-push', obj.imageName, obj.imageID, obj.tag);
+      ipcRenderer.send('do-image-push', obj.imageName.trim(), obj.imageID.trim(), obj.tag.trim());
```
Comment (mook-as) [inline src/components/Images.vue:246]: Sounds like using `ipcRenderer.invoke(…).then(…)` would be a lot cleaner?

**Review by mook-as — CHANGES_REQUESTED** [review]
This is getting to the point where we're holding too much state in the Vue component, and would work better to restructure the main process side to use `ipcMain.handle(…)` and return results instead.

File: background.ts (hunk 1 above)
Comment (ericpromislow) [reply background.ts:189]: I don't think so. It's all async. The Vue component isn't expecting a reply, but will act if it gets one.

File: src/components/Images.vue (hunk 2 above)
Comment (ericpromislow) [reply src/components/Images.vue:246]: For a UI, I prefer the current model of sending an event async, and then having another handler deal with any reply.

**Comment by ericpromislow** [comment]
> This is getting to the point where we're holding too much state in the Vue component, and would work better to restructure the main process side to use `ipcMain.handle(…)` and return results instead.

Why? The state has to live somewhere, right? OTOH, I could see a more elegant framework where replies to the VUE component/renderer consist of a JSON component that would be easily interpreted. The UI side could update its view without being too smart.

I would write an exploratory issue for that, rather than make it part of this one, which I intended to fix a few glitches.

**Review by mook-as — APPROVED** [review]
👍

**Pull request merged**